tests/test_mc_simulator.py

```diff
@@ -419,3 +445,154 @@ def test_MantarrayMCSimulator__handles_reads_of_size_greater_than_next_packet_in
 
     actual_2 = simulator.read(size=1)
     assert actual_2 == bytes(0)
+
+
+def test_MantarrayMCSimulator__raises_error_if_unrecognized_module_id_sent_from_pc(
+    mantarray_mc_simulator_no_beacon, mocker
+):
+    mocker.patch(
+        "builtins.print", autospec=True
+    )  # don't print all the error messages to console
+
+    _, _, _, _, simulator = mantarray_mc_simulator_no_beacon
+
+    dummy_timestamp = 0
+    dummy_packet_type = 1
+    test_module_id = 254
+    test_handshake = create_data_packet(
+        dummy_timestamp,
+        test_module_id,
+        dummy_packet_type,
+        DEFAULT_SIMULATOR_STATUS_CODE,
+    )
+
+    simulator.write(test_handshake)
+    time.sleep(QUEUE_CHECK_TIMEOUT_SECONDS)  # let input queue get populated
+
+    with pytest.raises(UnrecognizedSerialCommModuleIdError, match=str(test_module_id)):
+        invoke_process_run_and_check_errors(simulator)
+
+
+def test_MantarrayMCSimulator__raises_error_if_unrecognized_packet_type_sent_from_pc(
+    mantarray_mc_simulator_no_beacon, mocker
+):
+    mocker.patch(
+        "builtins.print", autospec=True
+    )  # don't print all the error messages to console
+
+    _, _, _, _, simulator = mantarray_mc_simulator_no_beacon
+
+    dummy_timestamp = 0
+    test_packet_type = 254
+    test_handshake = create_data_packet(
+        dummy_timestamp,
+        SERIAL_COMM_MAIN_MODULE_ID,
+        test_packet_type,
+        DEFAULT_SIMULATOR_STATUS_CODE,
+    )
+
+    simulator.write(test_handshake)
+    time.sleep(QUEUE_CHECK_TIMEOUT_SECONDS)  # let input queue get populated
+
+    with pytest.raises(UnrecognizedSerialCommPacketTypeError) as exc_info:
+        invoke_process_run_and_check_errors(simulator)
+    assert str(SERIAL_COMM_MAIN_MODULE_ID) in str(exc_info.value)
```

Comment:
`match` is a kwarg that the `pytest.raises` function accepts that you can put regular expressions or other things into to do this type of thing. there may be other ways it's more robust than this, or maybe just easier to read, not sure. but that's the pytest standard practice https://docs.pytest.org/en/stable/assert.html

Reply:
I usually use that, it just had problems detecting multiple int values for some reason